Hazel/src/Platform/Windows/WindowsWindow.cpp

```diff
@@ -10,6 +10,7 @@
 namespace Hazel {
 	
 	static bool s_GLFWInitialized = false;
+	static std::unordered_map<int16_t, unsigned int> s_KeyRepeatCounts;
```

Comment:
I really would like to see this as a static member of the input class...

###### My thoughts:
- Make this a private member
- Create a getter (to get the repeat count of a key)
- Create a setter (to set it to 0)
- Create a reset (to delete it upon release)
- Create an incrementer (to increase the repeat count, this could also act as a 2nd getter?)

###### Thinking about other platforms, code duplication
The increment/setting/resetting (aka use of the created functions) could be coded in the constructors of the `KeyPressedEvent` and the `KeyReleasedEvent` instead of the callback *(of each platform)*:
- `KeyPressedEvent` takes in a bool instead of an integer for distinguishing press/repeat
  - if not repeat then set to 0
  - if repeat then increment
- `KeyReleasedEvent` resets it to non-existing

Any other thoughts?

Reply:
Moving this to the `Input` class to allow polling is a good idea.

I'm not 100% sure what the use would be for the setter? My interpretation of what you suggest would be that you only have the ability to increment it or reset it (in terms of how you are able to modify it). Resetting it would remove it from the map, so there is no need to have a setter or to ever store "0 repeats" for a key. Unless you want to be able to keep the memory for the key and not have to potentially reallocate each time?

Should we store this directly in the Input class or delegate it to each Platform's input class? Some platforms might have this sort of counter already built in and keeping another one would be wasting some memory (although it's really not a lot of memory) in that case.

I don't think it would be that much code duplication to delegate it to each platform, but it is a valid point to consider still.

I'll push all the changes once we clear up the details.

Reply:
> I'm not 100% sure what the use would be for the setter? My interpretation of what you suggest would be that you only have the ability to increment it or reset it (in terms of how you are able to modify it). Resetting it would remove it from the map, so there is no need to have a setter or to ever store "0 repeats" for a key. Unless you want to be able to keep the memory for the key and not have to potentially reallocate each time?

My initial idea was a getter and a setter, and when I thought about the increment, the setter made sense for 0... if it is not existing, the key press count = 0, "incrementing" then would be the ++ operator...

But yeah, I think the getter function returning 0 when it doesn't exist would have the same effect, so no real reason for the setter, as you pointed out.

> Should we store this directly in the Input class or delegate it to each Platform's input class? Some platforms might have this sort of counter already built in and keeping another one would be wasting some memory (although it's really not a lot of memory) in that case.
>
> I don't think it would be that much code duplication to delegate it to each platform, but it is a valid point to consider still.

Hazel will use only 1 set of keycodes (and converter functions to go from one platform to the other), so I don't see why we can't store it in the input class? As the user will also use the hazel keycodes (as do the events), I don't see a specific reason to put it in the platform specific code?

> I'll push all the changes once we clear up the details.

Feel free to ask any questions you have 👍